### PR TITLE
docs/features: add installation docs for software-catalog and software-templates

### PR DESCRIPTION
**Pull request by Rugvip**
~WIP, need to add docs for the catalog as well, since the scaffolder depends on that~

Ready!

### CODE REVIEW

File: docs/features/software-catalog/descriptor-format.md [hunk 1]
```diff
@@ -316,10 +316,10 @@ group of people in an organizational structure.
...
 Describes the following entity kind:
 
-| Field                | Value                   |
-| -------------------- | ----------------------- |
-| `apiVersion`         | `backstage.io/v1alpha1` |
-| `Kind: Templatekind` | `Template`              |
+| Field            | Value                   |
+| ---------------- | ----------------------- |
+| `apiVersion`     | `backstage.io/v1alpha1` |
+| `Kind: Template` | `Template`              |
```
Comment (benjdlambert) [inline docs/features/software-catalog/descriptor-format.md:322]: this should just be `kind`

File: docs/features/software-templates/installation.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+Where `backendUrl` is the `backend.baseUrl` from config, i.e.
+`const backendUrl = config.getString('backend.baseUrl')`.
+
+This is all that is needed for the frontend part of the Scaffolder plugin to
+work!
+
+## Installing @backstage/plugin-scaffolder-backend
+
+The scaffolder backend should be installed in your `backend` package, which is
+created as a part of `@backstage/create-app`. To install the package, run:
```
Comment (angeliski) [inline docs/features/software-templates/installation.md:57]: I like that information, but the current template is [fully configured for that.](https://github.com/spotify/backstage/blob/master/packages/create-app/templates/default-app/packages/backend/src/plugins/scaffolder.ts).
Maybe we can add an alert for that, something like: "Your backend is already configured when using our create-app, but If you create a project from scratch, you can do follow that steps".

File: docs/features/software-templates/installation.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
+# Installing in your Backstage App
+
+The scaffolder plugin comes in two packages, `@backstage/plugin-scaffolder` and
+`@backstage/plugin-scaffolder-backend`. Each has their own installation steps,
+outlined below.
+
+The Scaffolder plugin also depends on the Software Catalog. Instructions for how
```
Comment (angeliski) [inline docs/features/software-templates/installation.md:7]: :heart: Really trick that

File: docs/features/software-templates/installation.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+For the scaffolder backend plugin to function, it needs a GitHub access token,
+and access to a running Docker daemon. You can create a GitHub access token
+[here](https://github.com/settings/tokens/new), select `repo` scope only. Full
+docs on creating private GitHub access tokens is available
+[here](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token).
+Note that the need for private GitHub access tokens will be replaced with GitHub
+Apps integration further down the line.
+
+The GitHub access token is passed along using the `GITHUB_ACCESS_TOKEN`
```
Comment (angeliski) [inline docs/features/software-templates/installation.md:171]: Here is nice an alert about the [org trick](https://github.com/spotify/backstage/issues/1881#issuecomment-671355877)

**Review by angeliski — COMMENTED** [review]
Really nice docs, I put some comments about trick paths when I got to running that process from scratch, I hope this help

File: docs/features/software-templates/installation.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+### Adding the Plugin to your `packages/app`
+
+Add the following entry to the head of your `packages/app/src/plugins.ts`:
+
+```ts
+export { plugin as ScaffolderPlugin } from '@backstage/plugin-scaffolder';
+```
+
+Add the following to your `packages/app/src/apis.ts`:
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply docs/features/software-templates/installation.md:32]: Yep we stumbled on this too, the docs were written with the assumption that we fix this the coming days 😅 

The work to fix the template is tracked here: https://github.com/spotify/backstage/issues/1843

We'll most likely document the setup of `apis.ts` as a part of the catalog installation docs for now, as the scaffolder depends on the catalog anyway.

File: docs/features/software-templates/installation.md (hunk 2 above)
Comment (Rugvip) [reply docs/features/software-templates/installation.md:57]: Yeah that's a good point. We wrote these docs as bit of an exercise and template for other installation docs, and simply to make sure we know of all things that are required. Should definitely mention that the backend already comes with this though.

File: docs/features/software-templates/installation.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (angeliski) [reply docs/features/software-templates/installation.md:32]: Amazing! PR #1843 really helps new projects :)

File: docs/features/software-catalog/index.md [hunk 6]
```diff
@@ -19,12 +19,19 @@ teams to jump between different infrastructure UI’s (and incurring additional
...
-## Using the Service Catalog
+## Getting Started
 
-TODO
+The Software Catalog is available to browse on the start page at `/`. If you've
+followed [Installing in your Backstage App](./installation.md) in your separate
+App or [Getting Started with Backstage](../../getting-started) for this repo,
+you should be able to browse the catalog at `http://localhost:3000`.
 
+<<<<<<< HEAD
```
Comment (benjdlambert) [inline docs/features/software-catalog/index.md:29]: me no likey

**Review by benjdlambert — CHANGES_REQUESTED** [review]

File: docs/features/software-catalog/index.md (hunk 6 above)
Comment (Rugvip) [reply docs/features/software-catalog/index.md]: me updatey

**Review by timja — APPROVED** [review]

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**